ADAL/src/broker/ios/ADBrokerNotificationManager.m

```diff
@@ -138,5 +138,9 @@ - (void)callbackCleanup:(NSNotification*)aNotification
     }
 }
 
+- (BOOL)hasCallback
+{
+    return _callbackForBroker != nil;
```

Comment:
Can it happen that someone else is calling copyAndClearCallback at the same time?

Reply:
This shouldn't happen since this method is being used on the main queue, but I added a lock, just to be consistent with other public methods.